src/Field/FileField.php

```diff
@@ -71,10 +72,12 @@ protected function init(): void
 
                 // mark new file as linked
                 if ($new) {
-                    $m->refModel($this->shortName)->loadBy('token', $new)->save(['status' => 'linked']);
+                    $m->refModel($this->shortName)->loadBy('token', $new)->save(['status' => File::STATUS_LINKED]);
                 }
             }
-        });
+        };
+        $this->onHookToOwnerEntity(Model::HOOK_AFTER_INSERT, $fx);
```

Comment:
just question - why after insert/update vs. after save hook?

please also confirm all your changes work with FK - ie. with DB setup script from this repo

Reply:
DB setup script works.

Because related records should be deleted before deleting parent record (source_file_id), but physical files should be deleted only after successful delete of DB record because of data integrity. Otherwise file gets deleted even in case DB record can't be deleted for some reason. DB rolls back, but physical file will still be deleted and can't "roll back".

Reply:
I have opened https://github.com/atk4/filestore/issues/35 - we must allow transaction to fail without any side effect.

Reply:
Thanks. And about AfterSave hook the trick is that AfterSave do not have access to dirty fields array anymore because of this https://github.com/atk4/data/blob/develop/src/Model.php#L1568 
That's why I use AfterInsert and AfterUpdate here.

Reply:
why we need dirty fields?

Reply:
to have access to previous file token and be able to delete it https://github.com/atk4/filestore/pull/34/files#diff-7dbb911d7a6858537e5799a3088ab37b99f98a02d613503dc350be285cb252b1R70